workflows/prognostic_run_diags/fv3net/artifacts/report_search.py

```diff
@@ -104,3 +104,34 @@ def _insert_public_domain(url) -> str:
             return url
         else:
             raise ValueError(f"Public domain unknown for url {url}.")
+
+
+def main(args):
+    if args.write:
+        index = ReportIndex()
+        index.compute(args.reports_url)
+        index.dump(os.path.join(args.reports_url, "index.json"))
+    index = ReportIndex.from_json(os.path.join(args.reports_url, "index.json"))
+    for link in index.public_links(args.url):
+        print(link)
```

Comment:
To enhance discoverability, it would be nice to refactor this as a function in `query`. That is where users (me) tend to look for functions like this.

Reply:
That seems a bit odd to me since the parser which points to this function is defined in `report_search.py`. I'm going to leave it as is.